rebasehelper/specfile.py

```diff
@@ -608,16 +608,19 @@ def _find_macros(s):
         def _expand_macros(s):
             """Expands all redefinable macros containing redefinable macros"""
             replace = []
+            macros = set()
             for macro, span in _find_macros(s):
                 value = _get_macro_value(macro)
                 if not value:
                     continue
-                rep = _expand_macros(value)
+                macros.add(macro)
+                rep, new_macros = _expand_macros(value)
+                macros |= new_macros
                 if _find_macros(rep):
                     replace.append((rep, span))
             for rep, span in reversed(replace):
                 s = s[:span[0]] + rep + s[span[1]:]
-            return s
+            return s, macros
```

Comment:
I think some extra documentation is needed here. With `_expand_macros()` returning a `str`, it's obvious what it is and what it does, but now it returns two values and that doesn't make much sense. At least the docstring should be extended.